### DISCUSSION TITLE
addFace() only for triangular faces? any function can handle quad or higher faces?

### DISCUSSION THREAD

**Discussion by blower05** [InteractiveComputerGraphics/PositionBasedDynamics#139] [Q&A] [answered]
Prof. Bender,

I found if I input addFace() with vertices more than 3 are not allowed. It means I can't input face other than triangles?
When I tried the following, 
`mesh = pbd.IndexedFaceMesh()`
`mesh.addFace([1,2,4,3,5,6,7,1,12,4])`
getFaces() returned:
[1, 2, 4, 1, 2, 4]

Sometimes, it would also break and close the python IDE.
Should I triangulate all the faces before using addFace() or what I should use for quad face or higher?

Thanks

**Comment by janbender** [comment, marked as answer]
The library only supports triangle meshes. So the simple answer is that you have to triangulate your mesh.

**Reply by blower05** [reply]
Thank you for confirming this that I know what to do. Thanks!
